### PR TITLE
Replace parcel watcher with chokidar

### PR DESCRIPTION
**Pull request by AlexandreSi**
Following issues building Desktop app with parcel watcher (See https://github.com/4ian/GDevelop/pull/5920), this PR brings back chokidar.

It also reduces unnecessary redefinitions of the watcher.
Now editors have to register themselves in order to get the information about when a resource has been changed.

Note: It is done using `componentDidMount` and `componentWillUnmount` but it can replaced with a single `React.useEffect` with a tidying function when the component is transformed into a functional component.

### CODE REVIEW

File: newIDE/electron-app/app/LocalFilesystemWatcher.js [hunk 1]
```diff
@@ -12,31 +12,38 @@ const getNewSubscriptionId = () => {
...
+      // never ending.
+      fileWiseCallback
     )
-    .then(subscription => {
-      subscriptions[newSubscriptionId] = subscription;
-    });
+    .on('unlink', fileWiseCallback)
+    .on('add', fileWiseCallback);
+
+  subscriptionCancelers[newSubscriptionId] = () => watcher.unwatch(folderPath);
```
Comment (ClementPasteau) [inline newIDE/electron-app/app/LocalFilesystemWatcher.js:34]: just to check, this function is not async?

**Review by ClementPasteau — APPROVED** [review]
fine for now!

File: newIDE/electron-app/app/LocalFilesystemWatcher.js (hunk 1 above)
Comment (AlexandreSi) [reply newIDE/electron-app/app/LocalFilesystemWatcher.js:34]: Nope!

**Review by 4ian — APPROVED** [review]
We'll bring back `@parcel/watcher` later with https://github.com/4ian/GDevelop/pull/5920

**Pull request merged**